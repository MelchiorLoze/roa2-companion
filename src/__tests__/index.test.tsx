import { render } from '@testing-library/react-native';
import { Redirect } from 'expo-router';

import Index from '@/app/index';

jest.mock('expo-router');
const RedirectMock = jest.mocked(Redirect);

const renderComponent = () => {
  return render(<Index />);
};

describe('index', () => {
  it('matches the snapshot', () => {
    const tree = renderComponent().toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('redirects to sign-in page', () => {
    renderComponent();

    expect(RedirectMock).toHaveBeenCalledWith({ href: '/sign-in' }, {});
  });
});
