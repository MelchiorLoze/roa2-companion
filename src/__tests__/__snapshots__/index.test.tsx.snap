// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index matches the snapshot 1`] = `null`;
